libs/template/src/lib/core/utils/template-manager.ts

```diff
@@ -0,0 +1,62 @@
+import { EmbeddedViewRef, TemplateRef, ViewContainerRef } from '@angular/core';
+
+type RxTemplateName = 'rxNext' | 'rxComplete' | 'rxError' | 'rxSuspense';
+
+export class TemplateManager<T> {
```

Comment:
Could you switch to factory functions instead of a class?

Like we did with the reader aware thingy?

Reply:
@BioPhoton Sure, I can do that :) I'll add that as a part of future commits.